### DISCUSSION TITLE
Adaptive meshing on 3D?

### DISCUSSION THREAD

**Discussion by qlyan913** [firedrakeproject/firedrake#3242] [Firedrake support]
Hello,

The tutorial here:https://www.firedrakeproject.org/demos/netgen_mesh.py.html gives an example of adaptive FEM on 2d domain. But when I try to apply it on a 3D cube, it gives error as followings: 

In refineMarkedElements
raise NotImplementedError("No implementation for dimension other than 2.")

So, is it still possible to do adaptive meshing on 3D domain with Firedrake?

Thanks.

**Comment by connorjward** [comment]
@UZerbinati is the guy to ask about this.
